pkg/proc/disasm.go

```diff
@@ -16,14 +16,27 @@ const (
 	IntelFlavour
 )
 
+func (dbp *Process) Disassemble(g *G, startPC, endPC uint64) ([]AsmInstruction, error) {
+	if g == nil {
+		regs, _ := dbp.currentThread.Registers(false)
+		return Disassemble(dbp.currentThread, regs, dbp.breakpoints, &dbp.bi, startPC, endPC)
+	}
+
+	var regs Registers
+	thread := dbp.currentThread
+	if g.thread != nil {
+		thread = g.thread
+		regs, _ = g.thread.Registers(false)
+	}
+
+	return Disassemble(thread, regs, dbp.breakpoints, &dbp.bi, startPC, endPC)
+}
+
 // Disassemble disassembles target memory between startPC and endPC
 // If currentGoroutine is set and thread is stopped at a CALL instruction Disassemble will evaluate the argument of the CALL instruction using the thread's registers
 // Be aware that the Bytes field of each returned instruction is a slice of a larger array of size endPC - startPC
-func (thread *Thread) Disassemble(startPC, endPC uint64, currentGoroutine bool) ([]AsmInstruction, error) {
-	if thread.dbp.exited {
-		return nil, &ProcessExitedError{}
-	}
-	mem, err := thread.readMemory(uintptr(startPC), int(endPC-startPC))
+func Disassemble(memrw memoryReadWriter, regs Registers, breakpoints map[uint64]*Breakpoint, bi *BinaryInfo, startPC, endPC uint64) ([]AsmInstruction, error) {
```

Comment:
For the most barebones functionality, yes. But we are also marking breakpoint and PC positions and resolving the CALL argument whenever possible. In the future I would also like to resolve SP relative addresses to local variable names.